test/spec/EnrollWindowsHello_spec.js

```diff
@@ -0,0 +1,180 @@
+        router.enrollWindowsHello();
+        return tick(null);
+      })
+      .then(function () {
+        return {
+          router: router,
+          beacon: new Beacon($sandbox),
+          form: new Form($sandbox),
+          ac: authClient,
+          setNextResponse: setNextResponse
+        };
+      });
+    }
+
+    function emulateNotWindows() {
+      spyOn(webauthn, 'isAvailable').and.returnValue(false);
+      spyOn(webauthn, 'makeCredential');
+      spyOn(webauthn, 'getAssertion');
+
+      return tick();
+    }
+
+    function emulateWindows() {
+      spyOn(webauthn, 'isAvailable').and.returnValue(true);
+
+      spyOn(webauthn, 'makeCredential').and.callFake(function () {
+        var deferred = $.Deferred();
+        deferred.catch = deferred.fail;
+
+        deferred.resolve({
+          credential: {id: 'credentialId'},
+          publicKey: 'publicKey'
+        });
+
+        return tick(deferred);
+      });
+
+      return tick();
+    }
+
+    Expect.describe('Header & Footer', function () {
+      itp('displays the correct factorBeacon', function () {
+        return emulateNotWindows()
+        .then(setup)
+        .then(function (test) {
+          expect(test.beacon.isFactorBeacon()).toBe(true);
+          expect(test.beacon.hasClass('mfa-windows-hello')).toBe(true);
+        });
+      });
+      itp('has a "back" link in the footer', function () {
+        return emulateNotWindows()
+        .then(setup)
+        .then(function (test) {
+          Expect.isVisible(test.form.backLink());
+        });
+      });
+    });
+
+    Expect.describe('Enroll factor', function () {
+      itp('displays error if not Windows', function () {
+        return emulateNotWindows()
+        .then(setup)
+        .then(function (test) {
+          expect(test.form.hasErrorNotWindows()).toBe(true);
+          expect(test.form.submitButton().length).toBe(0);
+        });
+      });
+
+      itp('doesnt display error if Windows', function () {
+        return emulateWindows()
+        .then(setup)
+        .then(function (test) {
+          expect(test.form.hasErrorNotWindows()).toBe(false);
+          expect(test.form.submitButton().length).toBe(1);
+        });
+      });
+
+      itp('subtitle changes after submitting the form', function () {
+        return emulateWindows()
+        .then(setup)
+        .then(function (test) {
+          expect(test.form.subtitleText())
+          .toEqual('Click below to enroll Windows Hello as a second form of authentication');
+          test.form.submit();
+          expect(test.form.subtitleText()).toEqual('Please wait while Windows Hello is loading...');
+        });
+      });
+
+      itp('sends enroll request after submitting the form', function () {
+        return emulateWindows()
+        .then(setup)
+        .then(function (test) {
+          test.form.submit();
+          return tick();
+        })
+        .then(function () {
+          expect($.ajax.calls.count()).toBe(2);
```

Comment:
The first is the login request to `api/v1/authn` and tested by the `PrimaryAuth_spec`